doc/rtd/topics/tutorials/qemu.rst

```diff
@@ -0,0 +1,282 @@
+the instance.
+
+
+How does cloud-init use the IMDS?
+=================================
+
+The IMDS contains a private http webserver to each operating
+system instance launched. During early boot, cloud-init sets up network
+access and queries this webserver to gather configuration data. This allows
+cloud-init to configure your operating system while it boots.
+
+In this tutorial we emulate this workflow using Qemu and a simple python
+webserver. This workflow may be suitable for developing and testing cloud-init
+configurations prior to cloud deployments.
+
+
+How to use this tutorial
+========================
+
+In this tutorial each code block is to be copied and pasted directly
+into the terminal then executed. Omit the prompt ``$`` before each command.
+
+Each code block is preceded by a description of what the command does.
+
+
+Install Qemu
+============
+
+Install Qemu.
+
+.. code-block:: sh
+
+    $ sudo apt install qemu-system-x86
+
+If you are not using Ubuntu, you can visit Qemu's `install instructions`_ for
+additional information.
+
+
+Create a temporary directory
+============================
+
+This directory will store our cloud image and configuration files,
+``meta-data`` [add-link], ``vendor-data`` [add-link], and ``user-data``
+[add-link],
+
+This tutorial expects that you run all commands from this temporary
+directory. Failure to do so will result in an unconfigured virtual
+machine.
+
+Create a temporary directory and make it your current working directory with
+``cd``.
+
+.. code-block:: sh
+
+   $ mkdir temp
+   $ cd temp
+
+
+Download a cloud image
+======================
+
+Cloud images typically come with cloud-init pre-installed and configured to run
+on first boot. Users should not need to worry about installing cloud-init
+unless they are manually creating their own images. In this case we select the
+latest Ubuntu LTS_.
+
+Download the server image using ``wget``.
+
+.. code-block:: sh
+
+    $ wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img
+
+
+Define our user data
+====================
+
+Create the following file ``user-data``. This user-data cloud-config
+sets the password of the default user and sets it to never expire. For
+more details see this module_.
+
+Execute the following command, which creates a file named ``user-data`` with
+configuration data.
+
+.. code-block:: sh
+
+    $ cat << EOF > user-data
+    #cloud-config
+    password: password
+    chpasswd:
+      expire: False
+
+    EOF
+
+
+What is user data?
+==================
+
+Before moving forward, let's inspect our user data file.
+
+.. code-block:: sh
+
+   $ cat user-data
+
+You should see the following contents:
+
+.. code-block:: yaml
+
+    #cloud-config
+    password: password
+    chpasswd:
+      expire: False
+
+The first line starts with ``#cloud-config``, which tells cloud-init
+what kind of configuration is contained. The cloud-config config type uses yaml
```

Comment:
```suggestion
what kind of configuration is contained. The cloud-config config type use yaml
```

Reply:
I'm going to push back a little on this change request. Since this config "type" is singular wouldn't `use` make more sense here?

I'm happy to make changes for clarity, but I do think that the current grammar makes more sense (which leads me to think that something about this sentence was confusing).

Thoughts?

Reply:
You're quite right, grammatically your original wording is correct, I have no idea why I thought it wasn't - I think the double "config" might have made my brain short circuit for a hot minute. If there is an alternative way of phrasing that sentence without changing your meaning, it might be a good idea to change it. It's a bit confusing to look at.

Reply:
> If there is an alternative way of phrasing that sentence without changing your meaning, it might be a good idea to change it. It's a bit confusing to look at.

How is this wording? https://github.com/canonical/cloud-init/pull/1863/commits/8398443f79ca46ce25beee74cb7f0eba23046f49

Reply:
Much better! Clear, but still concise - good job +1